servant-client/src/Servant/Common/Req.hs

```diff
@@ -149,20 +152,39 @@ parseRequest url = liftM disableStatusCheck (parseUrl url)
 displayHttpRequest :: Method -> String
 displayHttpRequest httpmethod = "HTTP " ++ cs httpmethod ++ " request"
 
-type ClientM = ExceptT ServantError IO
+-- previously: 
+--type ClientM = ExceptT ServantError IO
+
+data ClientEnv
+  = ClientEnv
+  { manager :: Manager
+  , baseUrl :: BaseUrl
+  }
```

Comment:
It may be interesting to provide convenience functions:

```
defaultLocalHostEnv :: Port -> IO ClientEnv
defaultLocalHostEnv port = do
   mgr <- newManager defautlManagerSettings
   return $ ClientEnv mgr (BaseUrl Http "localhost" port "")

defaultHttpsClient :: String -> IO ClientEnv
defaultHttpsClient host = ...
```